app/src/main/java/com/islamversity/reyan/service/MyFirebaseMessagingService.kt

```diff
@@ -0,0 +1,101 @@
+
+
+    override fun onNewToken(token: String) {
+        Timber.d("Notification : Refreshed token: $token")
+        sendFCMTokenToServer(token)
+    }
+
+    private fun sendFCMTokenToServer(token: String) {
+        // Cache.set(CacheKeys.Firebase.FCM_TOKEN_FIREBASE, token)
+        Timber.d("Notification : sendRegistrationTokenToServer($token)")
+    }
+
+    override fun onMessageReceived(remoteMessage: RemoteMessage) {
+        Timber.d("Notification : From: ${remoteMessage.from}")
+
+        val data  = remoteMessage.data
+
+        // Check if message contains a data payload.
+        if (data.isNotEmpty()) {
+            Timber.d("Notification : Message data payload: $data")
+
+            val bundle = Bundle()
+            for ((key, value) in data.entries) {
+                bundle.putString(key, value)
+            }
+
+            showNotification(
+                data["title"],
+                data["body"],
+                bundle
+            )
+        }
+
+        // Check if message contains a notification payload.
+        remoteMessage.notification?.let { notification ->
+            Timber.d("Notification : 2-Message Notification Body: ${notification.body}")
+        }
+    }
+
+    private fun showNotification(title: String?, body: String?, dataBundle: Bundle) {
+
+        val intent = Intent(this, MainActivity::class.java).apply {
+            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
+        }
+        intent.putExtra(NOTIFICATION_DATA_KEY, dataBundle)
+        val pendingIntent: PendingIntent = PendingIntent.getActivity(
+            this,
+            0,
+            intent,
+            PendingIntent.FLAG_ONE_SHOT
+        )
+
+        val channelId = getString(R.string.default_notification_channel_id)
+        val defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
+
+        val notificationBuilder = NotificationCompat.Builder(this, channelId)
+            .setSmallIcon(R.mipmap.ic_launcher)
+            .setContentTitle(title)
+            .setContentText(body)
+            .setStyle(NotificationCompat.BigTextStyle().bigText(body))
+            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+            .setSound(defaultSoundUri)
+            .setContentIntent(pendingIntent)
+            .setAutoCancel(true)
+
+        val notificationManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        // Since android Oreo notification channel is needed.
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(
+                channelId,
+                "Channel human readable title",
+                NotificationManager.IMPORTANCE_DEFAULT
+            )
+            notificationManager.createNotificationChannel(channel)
+        }
+
+        notificationManager.notify(0 /* ID of notification */, notificationBuilder.build())
```

Comment:
We need different id for each notification to avoid overlapping

Reply:
you are right, but it will not happen since we have just new_version  type. please mention  how to provide the id ?

Reply:
`Random.nextInt()` would work just fine in our case